src/hotspot/share/cds/filemap.cpp

```diff
@@ -963,7 +1029,13 @@ bool FileMapInfo::validate_app_class_paths(int shared_app_paths_len) {
     int j = header()->app_class_paths_start_index();
     mismatch = check_paths(j, shared_app_paths_len, rp_array);
     if (mismatch) {
-      return classpath_failure("[APP classpath mismatch, actual: -Djava.class.path=", appcp);
+      unsigned int dumptime_prefix_len = header()->common_app_classpath_size();
+      unsigned int runtime_prefix_len = longest_common_app_classpath_len(shared_app_paths_len, rp_array);
```

Comment:
How about `header()->common_app_classpath_prefix_size()` and `longest_common_app_classpath_prefix_len`?

Reply:
Done.